activemodel/test/cases/attribute_set_test.rb

```diff
@@ -279,6 +279,16 @@ def builder.build_from_database(values = {}, additional_types = {})
       assert_not_equal attributes2, attributes3
     end
 
+    test "==(other) is safe to use with any instance" do
+      attributes = AttributeSet.new({})
+
+      assert_equal false, attributes == nil
+      assert_equal false, attributes == BasicObject.new
+      assert_equal false, attributes == Object.new
+      assert_equal false, attributes == 1
+      assert_equal true, attributes == attributes
+    end
```

Comment:
```suggestion
    test "comparison for equality is safe to use with any instance" do
      attributes = AttributeSet.new({})

      assert_equal false, attributes == nil
      assert_equal false, attributes == 1
      assert_equal true, attributes == attributes
    end
```

Reply:
Could you please explain why we don't need to check it with BasicObject?

Reply:
And the same question about LazyAttributeSet

Reply:
We do not need to test against every possible object, just `nil`, something else and the same object (or object of the same class) would be enough.

Reply:
Fixed